targeting_terminal.py

```diff
@@ -0,0 +1,38 @@
+import pygame
+
+pygame.font.init()
+
+BLACK = 0, 0, 0
+BLUE = 0, 0, 255
+WHITE = 255, 255, 255
+EVENT_KEY_a = 97
+EVENT_KEY_z = 122
+EVENT_KEY_ENTER = 13
+EVENT_KEY_BACKSPACE = 8
+
+class TargetingTerminal():
+      
+    def __init__(self, DRAWING_SCALE):
+        self.current_text_ = ""
+        self.ui_font_ = pygame.font.SysFont("monospace", 10*DRAWING_SCALE)
+        self.DRAWING_SCALE_ = DRAWING_SCALE
+
+    def update(self , events):
+        for event in events:
+            textWidth = self.ui_font_.size(self.current_text_)[0]
+            if event.type == pygame.KEYDOWN:
+                if event.key == EVENT_KEY_ENTER:
+                    self.current_text_ = ""
+                if event.key == EVENT_KEY_BACKSPACE:
+                    self.current_text_ = self.current_text_[:-1]
+                elif event.key in range(EVENT_KEY_a, EVENT_KEY_z) and textWidth < 380:
+                    self.current_text_ += event.unicode.lower()
+
+    def draw_terminal(self, screen):
+        terminal_rect = pygame.Rect(10*self.DRAWING_SCALE_, screen.get_height()-(40*self.DRAWING_SCALE_), 400* self.DRAWING_SCALE_, 25*self.DRAWING_SCALE_)
```

Comment:
Since this is now a single class that is always instantiated, would it be more wise to do this calculation once on `__init__` and then keep re-using `terminal_rect` @cheunggg @enochtsang?

screen is not passed in on __init__, but maybe you can keep a bool that makes it do this calculation only on the first time calling "draw_terminal"?